lib/index.js

```diff
@@ -44,71 +48,106 @@ export default function runContentfulExport (usageParams) {
+      task: (ctx) => {
+        let successCount = 0
+        let warningCount = 0
+        let errorCount = 0
 
-      log.info('Downloading ' + response.assets.length + ' assets')
+        log.info('export', `Downloading ${ctx.data.assets.length} assets`)
 
-      return Promise.map(response.assets, (asset) => {
-        if (!asset.fields.file) {
-          log.warn('-> asset has no file(s)', jsonStringifySafe(asset))
-          warningCount++
-          return
-        }
-        const locales = Object.keys(asset.fields.file)
-        return Promise.mapSeries(locales, (locale) => {
-          const url = asset.fields.file[locale].url || asset.fields.file[locale].upload
-          return downloadAsset(url, opts.exportDir)
-          .then((downLoadedFile) => {
-            log.info('-> ' + downLoadedFile)
-            successCount++
-          })
-          .catch((error) => {
-            log.error('-> error downloading ' + url + ' => ' + error.message)
-            log.error(JSON.stringify(error, null, 2))
-            errorCount++
+        return Promise.map(ctx.data.assets, (asset) => {
+          if (!asset.fields.file) {
+            log.warn('-> asset has no file(s)', jsonStringifySafe(asset))
+            warningCount++
+            return
+          }
+          const locales = Object.keys(asset.fields.file)
+          return Promise.mapSeries(locales, (locale) => {
+            const url = asset.fields.file[locale].url || asset.fields.file[locale].upload
+            return downloadAsset(url, opts.exportDir)
+            .then((downLoadedFile) => {
+              log.info('export', '-> ' + downLoadedFile)
+              successCount++
+            })
+            .catch((error) => {
+              log.error('export', '-> error downloading ' + url + ' => ' + error.message)
+              log.error('export', JSON.stringify(error, null, 2))
+              errorCount++
+            })
           })
+        }, {
+          concurrency: 6
         })
-      }, {
-        concurrency: 6
-      })
-      .then(() => {
-        log.info('All ' + response.assets.length + ' assets downloaded.')
-        log.info('Successful file downloads: ' + successCount)
-        log.info('Download warnings: ' + warningCount)
-        log.info('Download errors: ' + errorCount)
-        return response
-      })
-    })
-    .then((response) => {
-      fs.existsSync(opts.exportDir) || fs.mkdirSync(opts.exportDir)
-      const responseFile = `${opts.exportDir}/contentful-export-${clients.source.spaceId}-${Date.now()}.json`
-      log.info('Writing space data to json file at : ' + responseFile)
-      fs.writeFileSync(responseFile, jsonStringifySafe(response, null, 4))
-      return response
+        .then(() => {
+          ctx.downloads = {
+            successCount,
+            warningCount,
+            errorCount
+          }
+        })
+      },
+      skip: () => !opts.downloadAssets
+    },
+    {
+      title: 'Writing data to file',
```

Comment:
We can add `skip: () => opts.saveFile` that Charli added in the previous [PR](https://github.com/contentful/contentful-export/pull/46) and remove [this](https://github.com/contentful/contentful-export/blob/master/lib/index.js#L101) check. It will look cleaner